### PR TITLE
Add new azure nodejs trigger samples

### PR DESCRIPTION
**Pull request by mydiemho**
- [x] storage blob
- [x] storage queue
- [x] service bus queue
- [x] service bus topic
- [x] event hub
- [x] timer

## Todos:

_**Note: Run `npm run test-ci` to run all validation checks on proposed changes**_

- [x] Write tests and confirm existing functionality is not broken.  
       **Validate via `npm test`**
- [x] Write documentation
- [x] Ensure there are no lint errors.  
       **Validate via `npm run lint-updated`**  
       _Note: Some reported issues can be automatically fixed by running `npm run lint:fix`_
- [x] Ensure introduced changes match Prettier formatting.  
       **Validate via `npm run prettier-check-updated`**  
       _Note: All reported issues can be automatically fixed by running `npm run prettify-updated`_
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

**_Is this ready for review?:_** YES  
**_Is it a breaking change?:_** NO

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (medikoo) [inline .eslintrc.json]: Was it committed in by mistake?

We keep configuration in `package.json` and it's important we test against syntax as supported in Node.js v6

So that file definitely should not be part of a PR :)

File: bin/serverless.js [hunk 2]
```diff
@@ -9,7 +9,6 @@ if (userNodeVersion >= 8) {
   const serverlessCli = require('@serverless/cli');
   if (serverlessCli.runningComponents()) {
     serverlessCli.runComponents();
-    return;
```
Comment (medikoo) [inline bin/serverless.js]: This change would be heavily breaking for us. I believe that's not intended commit.

File: lib/classes/Variables.js [hunk 3]
```diff
@@ -661,7 +661,6 @@ class Variables {
 
     // Process JS files
     if (fileExtension === 'js') {
-      // eslint-disable-next-line global-require, import/no-dynamic-require
```
Comment (medikoo) [inline lib/classes/Variables.js]: This should not be part of this PR

File: lib/plugins/aws/customResources/resources/utils.test.js [hunk 4]
```diff
@@ -1,6 +1,5 @@
 'use strict';
 
-// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (medikoo) [inline lib/plugins/aws/customResources/resources/utils.test.js]: Same with this one

File: package.json [hunk 5]
```diff
@@ -66,7 +66,9 @@
     "prettier-check": "prettier -c --ignore-path .gitignore \"**/*.{css,html,js,json,md,yaml,yml}\"",
     "prettier-check-updated": "pipe-git-updated --ext=css --ext=html --ext=js --ext=json --ext=md --ext=yaml --ext=yml -- prettier -c",
     "prettify": "prettier --write --ignore-path .gitignore \"**/*.{css,html,js,json,md,yaml,yml}\"",
-    "prettify-updated": "pipe-git-updated --ext=css --ext=html --ext=js --ext=json --ext=md --ext=yaml --ext=yml -- prettier --write"
```
Comment (medikoo) [inline package.json]: and this one

File: scripts/test-isolated.js [hunk 6]
```diff
@@ -133,4 +133,4 @@ const run = path => {
 };
 
 const limit = pLimit(processesCount);
-return initialSetupDeferred.then(() => Promise.all(paths.map(path => limit(() => run(path)))));
```
Comment (medikoo) [inline scripts/test-isolated.js]: And that one

File: docs/providers/azure/examples/event-hub/node/package.json [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  "keywords": [
+    "azure",
+    "serverless",
+    "event-hub",
+    "event-hub-trigger"
+  ],
+  "dependencies": {},
+  "devDependencies": {
+    "azurite": "^2.7.0",
+    "serverless-azure-functions": ">= 1.0.0-32"
```
Comment (pmuens) [inline docs/providers/azure/examples/event-hub/node/package.json]: We might want to update this version to be less specific.

File: docs/providers/azure/examples/hello-world/node/package.json [hunk 8]
```diff
@@ -1,11 +1,18 @@
...
-  "dependencies": {
-    "serverless-azure-functions": "^0.1.0"
+  "keywords": [
+    "azure",
+    "serverless",
+    "http-trigger"
+  ],
+  "dependencies": {},
+  "devDependencies": {
+    "serverless-azure-functions": ">= 1.0.0-32"
```
Comment (pmuens) [inline docs/providers/azure/examples/hello-world/node/package.json]: See comment above.

File: docs/providers/azure/examples/service-bus-queue/node/package.json [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+    "azure",
+    "serverless",
+    "service-bus",
+    "service-bus-queue",
+    "service-bus-queue-trigger"
+  ],
+  "dependencies": {},
+  "devDependencies": {
+    "azurite": "^2.7.0",
+    "serverless-azure-functions": ">= 1.0.0-32"
```
Comment (pmuens) [inline docs/providers/azure/examples/service-bus-queue/node/package.json]: See comment above.

File: docs/providers/azure/examples/service-bus-topic/node/package.json [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+    "azure",
+    "serverless",
+    "service-bus",
+    "service-bus-topic",
+    "service-bus-topic-trigger"
+  ],
+  "dependencies": {},
+  "devDependencies": {
+    "azurite": "^2.7.0",
+    "serverless-azure-functions": ">= 1.0.0-32"
```
Comment (pmuens) [inline docs/providers/azure/examples/service-bus-topic/node/package.json]: See comment above.

File: docs/providers/azure/examples/storage-blob/node/package.json [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+  "keywords": [
+    "azure",
+    "serverless",
+    "storage-blob",
+    "storage-blob-trigger"
+  ],
+  "dependencies": {},
+  "devDependencies": {
+    "azurite": "^2.7.0",
+    "serverless-azure-functions": ">= 1.0.0-32"
```
Comment (pmuens) [inline docs/providers/azure/examples/storage-blob/node/package.json]: See comment above.

File: docs/providers/azure/examples/storage-queue/node/package.json [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+  },
+  "keywords": [
+    "azure",
+    "serverless",
+    "storage-queue"
+  ],
+  "dependencies": {},
+  "devDependencies": {
+    "azurite": "^2.7.0",
+    "serverless-azure-functions": ">= 1.0.0-32"
```
Comment (pmuens) [inline docs/providers/azure/examples/storage-queue/node/package.json]: See comment above.

File: docs/providers/azure/examples/timer/node/package.json [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+  "keywords": [
+    "azure",
+    "serverless",
+    "timer",
+    "timer-trigger"
+  ],
+  "dependencies": {},
+  "devDependencies": {
+    "azurite": "^2.7.0",
+    "serverless-azure-functions": ">= 1.0.0-32"
```
Comment (pmuens) [inline docs/providers/azure/examples/timer/node/package.json]: See comment above.

File: .eslintrc.json (hunk 1 above)
Comment (mydiemho) [reply .eslintrc.json]: I will move this into the samples directory as this was created to support `async/await` syntax in the sample code

File: bin/serverless.js (hunk 2 above)
Comment (mydiemho) [reply bin/serverless.js]: will undo this as this was change after adding `.eslintrc.js`

File: lib/classes/Variables.js (hunk 3 above)
Comment (mydiemho) [reply lib/classes/Variables.js]: will undo this as this was change after adding `.eslintrc.js`

File: lib/plugins/aws/customResources/resources/utils.test.js (hunk 4 above)
Comment (mydiemho) [reply lib/plugins/aws/customResources/resources/utils.test.js]: will undo this as this was change after adding `.eslintrc.js`

**Comment by mydiemho** [comment]
@medikoo  @pmuens  should I move everything to [events](https://serverless.com/framework/docs/providers/azure/events/queuestorage/) instead.  In the current doc, events is more useful than the `examples` section.  

Should the examples also include instruction on how to test it? What's the difference between the `events` and `examples` section?

**Comment by pmuens** [comment]
Thanks for following-up @mydiemho :+1:

> @medikoo @pmuens should I move everything to events instead. In the current doc, events is more useful than the examples section.

> Should the examples also include instruction on how to test it? What's the difference between the events and examples section?

IMHO we should move these examples to the [`serverless/examples`](https://github.com/serverless/examples) repository and link from the event docs to that repo. This way there's only one source of truth and the examples won't get out of sync.

The main difference is that the events section just describes what one can do (but doesn't go into the implementation details on how to accomplish that goal). It's basically prose + some example config, but no fully-fledged application / service one can deploy. That's actually the purpose of the `serverless/examples` repository. In there we have a whole bunch of ready-to-deploy / ready-to-test services one can install via `serverless create` or `serverless install`.

**Comment by mydiemho** [comment]
> IMHO we should move these examples to the [`serverless/examples`](https://github.com/serverless/examples) repository and link from the event docs to that repo. This way there's only one source of truth and the examples won't get out of sync.

Oh I wasn't aware of the examples repo, I agree that it's a much better place.  Will fix this up next week and submit the changes.

**Comment by pmuens** [comment]
> Oh I wasn't aware of the examples repo, I agree that it's a much better place. Will fix this up next week and submit the changes.

Awesome! Thanks for working on this @mydiemho 👍

@eahefnawy is the maintainer of the `examples` repo and he's more than happy to merge the new Azure examples! Let us know if you need a helping hand here!

**Comment by mydiemho** [comment]
@pmuens  @medikoo  going to close this PR and create a pr in the examples repo.

Did open a separate PR to update existing docs for Azure: #6664

**Comment by fearphage** [comment]
@mydiemho Did this ever get moved/reopened in [serverless/examples](https://github.com/serverless/examples)? If so could you link it here.

**Comment by mydiemho** [comment]
@fearphage, haven’t had a chance to move it over yet.  Will let you know once it’s ready
